src/vit-servicing-station/vit-servicing-station-tests/src/tests/raw_snapshot.rs

```diff
@@ -0,0 +1,361 @@
+        .with_tag("fund9")
+        .build();
+
+    rest_client.put_raw_snapshot(&second_raw_snapshot).unwrap();
+
+    let assigner = RepsVotersAssigner::new(
+        first_raw_snapshot.content.direct_voters_group.unwrap(),
+        first_raw_snapshot.content.representatives_group.unwrap(),
+    );
+
+    let first_snapshot = Snapshot::from_raw_snapshot(
+        first_raw_snapshot.content.snapshot,
+        first_raw_snapshot.content.min_stake_threshold,
+        first_raw_snapshot.content.voting_power_cap,
+        &assigner,
+    )
+    .unwrap()
+    .to_full_snapshot_info();
+
+    for (idx, entry) in first_snapshot.iter().enumerate() {
+        let voting_power = VotingPower::from(entry.clone());
+        let voter_info = rest_client
+            .voter_info(&first_raw_snapshot.tag, &entry.hir.voting_key.to_hex())
+            .unwrap();
+        assert_eq!(
+            vec![voting_power.clone()],
+            voter_info.voter_info,
+            "wrong data for entry idx: {}",
+            idx
+        );
+        assert_eq!(
+            first_raw_snapshot.content.update_timestamp, voter_info.last_updated,
+            "wrong timestamp for entry idx: {}",
+            idx
+        );
+
+        let voter_info = rest_client
+            .voter_info(&second_raw_snapshot.tag, &entry.hir.voting_key.to_hex())
+            .unwrap();
+        assert_eq!(
+            vec![voting_power],
+            voter_info.voter_info,
+            "wrong data for entry idx: {}",
+            idx
+        );
+        assert_eq!(
+            second_raw_snapshot.content.update_timestamp, voter_info.last_updated,
+            "wrong timestamp for entry idx: {}",
+            idx
+        );
+    }
+}
+
+#[test]
+pub fn import_malformed_raw_snapshot() {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, data) = quick_start(&temp_dir).unwrap();
+    let rest_client: RawRestClient = server.rest_client_with_token(&data.token_hash()).into();
+
+    let raw_snapshot = RawSnapshot::default();
+    let mut content = serde_json::to_string(&raw_snapshot.content).unwrap();
+    content.pop();
+    assert!(rest_client
+        .put_raw_snapshot(&raw_snapshot.tag, content)
+        .unwrap()
+        .status()
+        .is_client_error());
+}
+
+#[test]
+pub fn import_big_raw_snapshot() {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, data) = quick_start(&temp_dir).unwrap();
+    let mut rest_client = server.rest_client_with_token(&data.token_hash());
+    rest_client.set_timeout(Duration::new(600, 0));
+
+    let raw_snapshot = RawSnapshotBuilder::default()
+        .with_tag("big")
+        .with_voting_registrations_count(100_000)
+        .build();
+
+    rest_client.put_raw_snapshot(&raw_snapshot).unwrap();
+
+    let assigner = RepsVotersAssigner::new(
+        raw_snapshot.content.direct_voters_group.unwrap(),
+        raw_snapshot.content.representatives_group.unwrap(),
+    );
+
+    let snapshot = Snapshot::from_raw_snapshot(
```

Comment:
I also think this can be nicely wrapped with some method. We don't need to know at this level of test how we construct snapshot (as we don;t check parameter influence on outcome).

here maybe and extension method would be better:
```
pub trait RawSnapshotExtension {
   fn into_full_snapshot(self, assigner: &RepsAssigner) -> Result<Snapshot,Error> ;
}


impl RawSnapshotExtension  for RawSnapshot {
  fn into_full_snapshot(self, assigner: &RepsAssigner) -> Result<Snapshot,Error>{
Ok(Snapshot::from_raw_snapshot(
         self..content.snapshot,
        self.content.min_stake_threshold,
        self.content.voting_power_cap,
        assigner,
    )?
    .to_full_snapshot_info())
}
```

and then use it:

`raw_snapshot.into_full_snapshot(&assigner),unwrap()`

Reply:
Done